qiskit/transpiler/passes/calibration/pulse_gates.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Add user provided calibrations."""
+
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.pulse.instruction_schedule_map import InstructionScheduleMap, CalibrationPublisher
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class PulseGates(TransformationPass):
```

Comment:
This may have some overlap with the following class: https://github.com/Qiskit/qiskit-terra/blob/9b48476cbaf5c4742f67e1ff6ce2546ba5fad01f/qiskit/transpiler/passes/scheduling/calibration_creators.py#L37-L69
is it possible to reduce this overlap by having a common base class?

Reply:
Good suggestion. see https://github.com/Qiskit/qiskit-terra/pull/6759/commits/94210f79ea4622f8f6e168064fca5097e3504415